src/org/mozilla/javascript/JavaMembers.java

```diff
@@ -918,3 +740,988 @@ public Object getDefaultValue(Class<?> hint)
     Field field;
     Object javaObject;
 }
+
+class JavaMembersNew extends JavaMembers
+{
+    JavaMembersNew(Scriptable scope, Class<?> cl)
```

Comment:
Also, class names like "JavaMembersNew" and "JavaMembersOld" aren't necessarily easy to understand. Is one class intended to be "with cache" and the other "without cache?" It might make it easier to call that out in the name.

Reply:
> Also, class names like "JavaMembersNew" and "JavaMembersOld" aren't necessarily easy to understand. Is one class intended to be "with cache" and the other "without cache?" It might make it easier to call that out in the name.

I perfer "JavaMembersNew" to "JavaMembers4CacheAndLazyWay" and "JavaMembersOld" to "JavaMembers4NoCacheNoLazy", is that ok ?